### ISSUE TITLE
megacli plugin on FreeBSD does not work without patching

### ISSUE THREAD

**Issue by samm-git** [netdata/netdata#7107]
##### Bug report summary

File `megacli.chart.py` does not work on FreeBSD without patches and needs to be more configurable

##### OS / Environment

FreeBSD bunny 12.0-RELEASE-p10 FreeBSD 12.0-RELEASE-p10 GENERIC  amd64

##### Netdata version (ouput of `netdata -V`)

netdata v1.17.1

##### Component Name

python.d/megacli

##### Steps To Reproduce

1. Install in FreeBSD with MFI adapter
1. Add rule to sudo )
1. Start netdata

It will print

```
2019-10-14 21:46:27: python.d ERROR: megacli[megacli] : sudo: a password is required
2019-10-14 21:46:27: python.d INFO: plugin[main] : megacli[megacli] : check failed
```

Line from sudo:
`netdata ALL= NOPASSWD: /usr/local/sbin/MegaCli`

##### Expected behavior

Working plugin

##### Additional information:

- Binary on FreeBSD is called `/usr/local/sbin/MegaCli` (see `pkg info -l megacli`). But changing `self.m = find_binary('MegaCli')` is not enough to fix the issue.
- Sudo check failed, as `sudo -v -n` with this config will give `sudo: a password is required`, si i had to replace it with `sudo -V`

I got it working with the following patch:

```diff
--- megacli.chart.py.old	2019-10-15 22:18:50.433634000 +0200
+++ megacli.chart.py	2019-10-15 22:38:52.235694000 +0200
@@ -163,8 +163,9 @@
 class Megacli:
     def __init__(self):
         self.s = find_binary('sudo')
-        self.m = find_binary('megacli')
-        self.sudo_check = [self.s, '-n', '-v']
+        self.m = find_binary('MegaCli')
+        self.sudo_check = [self.s, '-V']
         self.disk_info = [self.s, '-n', self.m, '-LDPDInfo', '-aAll', '-NoLog']
         self.battery_info = [self.s, '-n', self.m, '-AdpBbuCmd', '-a0', '-NoLog']
```

I decided to report it to get fixed this in the upstream, if possible

**Comment by ilyam8** [comment]
@hi @samm-git 

> Sudo check failed, as sudo -v -n with this config will give sudo: a password is required, si i had to replace it with sudo -V

I am not clear about it. 

> > -v, --validate
                 Update the user's cached credentials, authenticating the user if
                 necessary.  For the sudoers plugin, this extends the sudo timeout
                 for another 15 minutes by default, but does not run a command. 
                 Not all security policies support cached credentials.


> If your user is only allowed to run specific commands, this command will work, indicating you are allowed to run something with different privileges.


It looks like it is exactly what we need.

 - case `netdata is not in sudoers`

```
netdata@debian10:/home/test$ sudo -n -v 1>/dev/null
Sorry, user netdata may not run sudo on debian10.
```

 - case `netdata is in sudoers`

```
netdata@debian10:/home/test$ sudo -n -v
netdata@debian10:/home/test$
```

**Comment by ilyam8** [comment]
> sudo -V

is not really helpful, the idea is to check if user is able to run any sudo command

**Comment by samm-git** [comment]
@ilyam8 I provided line in sudoers for netdata already:

```
root@bunny:~ # grep netdata /usr/local/etc/sudoers
netdata ALL= NOPASSWD: /usr/local/sbin/MegaCli
```
And some details:
```
root@bunny:~ # sudo -u netdata -s
> id
uid=302(netdata) gid=302(netdata) groups=302(netdata),62(proxy),65534(nobody)
> sudo -n -v
sudo: a password is required
> sudo /usr/local/sbin/MegaCli
Fatal error - Command Tool invoked with wrong parameters
Exit Code: 0x01
>
```
As you could see - sudo allows to run MegaCli but `-n -v` is not working. So i think that this check is invalid.

**Comment by samm-git** [comment]
I think if you want to have a valid check - do `sudo -n /usr/local/sbin/MegaCli -v`. As i do not see any reason why sudo should allow to run **any** command.

**Comment by ilyam8** [comment]
No, i didnt mean run any command. I mean at least something.

> If your user is only allowed to run specific commands, this command will work, indicating you are allowed to run something with different privileges.

**Comment by samm-git** [comment]
@ilyam8 as you could see - it does not. Also doc states that it could not work:

```
     -v, --validate
                 Update the user's cached credentials, authenticating the user
                 if necessary.  For the sudoers plugin, this extends the sudo
                 timeout for another 5 minutes by default, but does not run a
                 command.  Not all security policies support cached
                 credentials.
```

**Comment by samm-git** [comment]
And again - i do not see any point to have such abstract check. If you want to check that sudo binary could be executed - `sudo -V` is enough. If your goal is to check that it can run megacli tool - `sudo -n <megacli bin> -v`. Everything else is not important for that usecase IMHO.

**Comment by samm-git** [comment]
Actually message you quoted is some stackexchange reply, w/o any kind of proofs and is not inline with what manual tells us.

**Comment by samm-git** [comment]
@ilyam8 btw, `sudo -l` works as expected:

```
> sudo -l
User netdata may run the following commands on bunny:
    (root) NOPASSWD: /usr/local/sbin/MegaCli
>
```

**Comment by ilyam8** [comment]
I do not argue it works everywhere, i see it doesnt work for you and i saw it earlier

> https://github.com/netdata/netdata/issues/6495#issuecomment-537447477

> sudo -n -v

worked for me on deb9-10 (and i think on 8 too, dont remember if i tested it there)

**Comment by ilyam8** [comment]
yes, 

> sudo -n -l

does the job w eneed

**Comment by samm-git** [comment]
@ilyam8 so i think it should be replaced to something more reliable or removed completely. As it is now just preventing plugin to work on a standard FreeBSD host, and probably not only :)

**Comment by samm-git** [comment]
After the changes i provided it works as expected :)
<img width="854" alt="Screenshot 2019-10-15 at 23 24 55" src="https://user-images.githubusercontent.com/1388875/66871027-0ee35d00-efa3-11e9-95f1-1783da2d2d44.png">

**Comment by samm-git** [comment]
@ilyam8 jfyi - i tested with `self.sudo_check = [self.s, '-n', '-l']` and it works as expected for me

**Comment by ilyam8** [comment]
@samm-git 

thx for reporting, i think the bug is fixed in #7108
